ocs_ci/deployment/deployment.py

```diff
@@ -652,3 +667,131 @@ def create_catalog_source(image=None, ignore_upgrade=False):
     )
     # Wait for catalog source is ready
     catalog_source.wait_for_state("READY")
+
+
+def setup_local_storage():
+    """
+    Setup the necessary resources for enabling local storage.
+
+    """
+    # Get the worker nodes
+    workers = get_typed_nodes(node_type='worker')
+    worker_names = [worker.name for worker in workers]
+    worker_names_str = " ".join(worker_names)
+    logger.debug("Workers: %s", workers)
+
+    # Label the worker nodes
+    logger.info("Labeling worker nodes")
+    run_cmd(
+        f"oc label nodes {worker_names_str} "
+        f"{constants.OPERATOR_NODE_LABEL}"
+    )
+
+    logger.info("Retrieving local-storage-operator data from yaml")
+    lso_data = list(templating.load_yaml(
+        constants.LOCAL_STORAGE_OPERATOR, multi_document=True
+    ))
+    # Update local-storage-operator subscription data with channel
+    for data in lso_data:
+        if data['kind'] == 'Subscription':
+            data['spec']['channel'] = config.ENV_DATA.get('ocs_version')
+
+    # Create temp yaml file and create local storage operator
+    logger.info(
+        "Creating temp yaml file with local-storage-operator data:\n %s",
+        lso_data
+    )
+    lso_data_yaml = tempfile.NamedTemporaryFile(
+        mode='w+', prefix='local_storage_operator', delete=False
+    )
+    templating.dump_data_to_temp_yaml(
+        lso_data, lso_data_yaml.name
+    )
+    logger.info("Creating local-storage-operator")
+    run_cmd(f"oc create -f {lso_data_yaml.name}")
+
+    local_storage_operator = ocp.OCP(
+        kind=constants.POD, namespace='local-storage'
+    )
+    assert local_storage_operator.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=constants.LOCAL_STORAGE_OPERATOR_LABEL,
+        timeout=600
+    ), "Local storage operator did not reach running phase"
+
+    # Retrieve NVME device path ID for each worker node
+    dev_paths = []
+    for worker in worker_names:
+        logger.info("Retrieving device path for node: %s", worker)
+        cmd = (
+            f"oc debug nodes/{worker} "
+            f"-- chroot /host ls -la /dev/disk/by-id/"
+        )
+        out = run_cmd(cmd)
+        out_lines = out.split('\n')
+        pattern = 'nvme-nvme.1d0f-'
+        nvme_line = [line for line in out_lines if pattern in line][0]
```

Comment:
This will take the OS device too. i think you can look in the pattern for EC2, this will give you the Free devices only.

Reply:
Here's the full output I get from this command:

```
drwxr-xr-x. 2 root root 340 Mar 25 15:27 .
drwxr-xr-x. 8 root root 160 Mar 25 15:27 ..
lrwxrwxrwx. 1 root root  10 Mar 25 15:27 dm-name-coreos-luks-root-nocrypt -> ../../dm-0
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-Amazon_EC2_NVMe_Instance_Storage_AWS6428258570A1B4DC5 -> ../../nvme1n1
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-Amazon_EC2_NVMe_Instance_Storage_AWSB9CF180FB5F23849F -> ../../nvme2n1
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-Amazon_Elastic_Block_Store_vol088bf4676b265b3ba -> ../../nvme0n1
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-Amazon_Elastic_Block_Store_vol088bf4676b265b3ba-part1 -> ../../nvme0n1p1
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-Amazon_Elastic_Block_Store_vol088bf4676b265b3ba-part2 -> ../../nvme0n1p2
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-Amazon_Elastic_Block_Store_vol088bf4676b265b3ba-part3 -> ../../nvme0n1p3
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-Amazon_Elastic_Block_Store_vol088bf4676b265b3ba-part4 -> ../../nvme0n1p4
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-nvme.1d0f-4157533634323832353835373041314234444335-416d617a6f6e20454332204e564d6520496e7374616e63652053746f72616765-00000001 -> ../../nvme1n1
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-nvme.1d0f-4157534239434631383046423546323338343946-416d617a6f6e20454332204e564d6520496e7374616e63652053746f72616765-00000001 -> ../../nvme2n1
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-nvme.1d0f-766f6c3038386266343637366232363562336261-416d617a6f6e20456c617374696320426c6f636b2053746f7265-00000001 -> ../../nvme0n1
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-nvme.1d0f-766f6c3038386266343637366232363562336261-416d617a6f6e20456c617374696320426c6f636b2053746f7265-00000001-part1 -> ../../nvme0n1p1
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-nvme.1d0f-766f6c3038386266343637366232363562336261-416d617a6f6e20456c617374696320426c6f636b2053746f7265-00000001-part2 -> ../../nvme0n1p2
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-nvme.1d0f-766f6c3038386266343637366232363562336261-416d617a6f6e20456c617374696320426c6f636b2053746f7265-00000001-part3 -> ../../nvme0n1p3
lrwxrwxrwx. 1 root root  15 Mar 25 15:27 nvme-nvme.1d0f-766f6c3038386266343637366232363562336261-416d617a6f6e20456c617374696320426c6f636b2053746f7265-00000001-part4 -> ../../nvme0n1p4
```

The devices with EC2 in them:
```
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-Amazon_EC2_NVMe_Instance_Storage_AWS6428258570A1B4DC5 -> ../../nvme1n1
lrwxrwxrwx. 1 root root  13 Mar 25 15:27 nvme-Amazon_EC2_NVMe_Instance_Storage_AWSB9CF180FB5F23849F -> ../../nvme2n1
```

The ID my code is grabbing:
```
nvme-nvme.1d0f-4157533634323832353835373041314234444335-416d617a6f6e20454332204e564d6520496e7374616e63652053746f72616765-00000001
```

The device-id I'm grabbing (and the one similar to it that links to `nvme2n1`) seem to the same as the two that have EC2 in them. How can I differentiate between the OS device and the other from this output?

Reply:
I think can use `blkid` to figure out fs on partitions on skip them, ceph-disk has used variety of things before, it is quite complex for all scenarios but for us its a good start.

Reply:
I see what @AviadP means now. If we use a pattern like `nvme-Amazon_EC2_NVMe_Instance_Storage` then we will only get the two free NVME devices and not the OS device in addition to the free ones. What I have works because the ones we want are first in the list but if the list only contains the free devices that is better.

Reply:
I'm now using `Amazon_EC2_NVMe_Instance_Storage` as the pattern we use to find the free NVME device paths. I also changed the logic to add both drives this was brought up as something we should be doing in our platform testing syncup.